app/src/main/java/org/fossasia/openevent/general/event/EventDao.kt

```diff
@@ -37,5 +37,5 @@ interface EventDao {
     fun getFavoriteEventWithinIds(ids: List<Long>): Single<List<Long>>
 
     @Query("SELECT * from Event WHERE eventTopic = :topicId")
-    fun getAllSimilarEvents(topicId: Long): Flowable<List<Event>>
+    fun getAllSimilarEvents(topicId: Long): Single<List<Event>>
```

Comment:
Why?

Reply:
Why do we need continuous streaming from database?
Even though we should remove database call for similar events, it is always empty and go for api call.

Reply:
The only reason we are using the database is because we need to observe changes without requerying

Reply:
Oh okay, and this query isn't working, it always returns 0 result.

Reply:
Then diagnose why